[Ramesh Kumar Sahana Avantika] iP

SahanaAvantika opened the pull request:
# Ben Chatbot

Ben is a chatbot that helps you organize your tasks
through the power of the Omnitrix

> “The best way to be prepared for the future is to take control of the present.” 
> – Ben Tennyson

**Ben frees your mind from having to remember things you need to do. It's:**
- Text-based
- Easy to learn
- ~~Fast~~ Super Fast to use

**All you need to do is:**

1. Download it from [here](https://github.com/SahanaAvantika/ip). 
2. Double-click it. 
3. Add your tasks. 
4. Let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [X] Managing Tasks 
- [X] Updating Tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

**Warning:** Keep the `src\main\java` folder as the root folder for Java files (i.e., don't rename those folders or move Java files to another folder outside of this folder path), as this is the default location some tools (e.g., Gradle) expect to find Java files.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SahanaAvantika committed:
Week 2 ip completed

SahanaAvantika committed:
week 2 ip completed

SahanaAvantika committed:
add level 7

SahanaAvantika committed:
format string to date

SahanaAvantika committed:
add A-MoreOOP

SahanaAvantika committed:
add Command

SahanaAvantika committed:
add packages

SahanaAvantika committed:
Merge branch 'add-gradle-support'

SahanaAvantika committed:
add gradle

SahanaAvantika committed:
add gradle support

SahanaAvantika committed:
add JUnit test

SahanaAvantika committed:
add A-Jar

SahanaAvantika committed:
add JavaDoc

SahanaAvantika committed:
complete coding standard

SahanaAvantika committed:
Merge branch 'branch-A-CodingStandard'

SahanaAvantika committed:
add find feature

SahanaAvantika committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Task/Task.java

SahanaAvantika committed:
add Level 9

jniaorx commented on src/main/java/Commands/AddCommand.java:
```diff
@@ -0,0 +1,25 @@
+import Task.*;
+
+
+public class AddCommand extends Commands {
```
This class is pretty well-written! However, would it be better for you to use nouns instead for naming this class?

jniaorx commented on src/main/java/Commands/DeleteCommand.java:
```diff
@@ -0,0 +1,26 @@
+import Ui.Ui;
+import Task.*;
+
+public class DeleteCommand extends Commands {
```
Good job once again! However, as mentioned, would it be better for you to use nouns instead for naming of this class?

jniaorx commented on src/main/java/Ui/Ui.java:
```diff
@@ -0,0 +1,98 @@
+    /**
+     * Displays a task is marked done to the user.
+     */
+    public static void markDone(TaskList list, int i){
```
Good job on using verb for method!

jniaorx commented on src/main/java/Ui/Ui.java:
```diff
@@ -0,0 +1,98 @@
+     *
+     * @param tasks is all the tasks in the list with the input word given by the user.
+     */
+    public static void find(ArrayList<Task> tasks){
```
You clearly distinguished between single-valued and multi-valued variables. Well done!

jniaorx commented on src/main/java/TaskList/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+     * @return ArrayList with the tasks that contain that des.
+     */
+    public ArrayList<Task> getTasks(String des){
+        ArrayList<Task> aptTask = new ArrayList<>();
```
Maybe it would be better if "aptTask" is in plural? This might help to better distinguish between single-valued and multi-valued variables.

jniaorx commented on src/main/java/Task/Deadlines.java:
```diff
@@ -0,0 +1,25 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadlines extends Task {
```
Good job on using noun for class naming!

jniaorx reviewed (commented):
Overall, the coding quality is pretty good! The namings are pretty self-explanatory and the lengths of the namings look good! There are only some mistakes made with the naming of classes where nouns should be used, and a minor mistake regarding single-valued and multi-valued variables.

SahanaAvantika committed:
add GUI

SahanaAvantika committed:
update ReadMe

SahanaAvantika committed:
update GUI

SahanaAvantika committed:
Merge branch 'branch-Level-10'

SahanaAvantika committed:
Use assert feature

SahanaAvantika committed:
Edit to code quality

SahanaAvantika committed:
Merge branch 'master' into branch-A-CodeQuality

SahanaAvantika committed:
Merge pull request #1 from SahanaAvantika/branch-A-CodeQuality

Branch a code quality

SahanaAvantika committed:
Add update feature

SahanaAvantika committed:
Add error catching

SahanaAvantika committed:
Add intro message

SahanaAvantika committed:
The GUI closes without executing the "bye" command, which prevents the application from displaying the exit message or performing any necessary cleanup.

This results in an abrupt termination, making it unclear whether the application has properly shut down.

Let's ensure the "bye" command is processed when the window is closed. Modify Main to call MainWindow.bye() in setOnCloseRequest(), ensuring the exit message is displayed and any cleanup logic in Ben runs before termination.

Using bye() instead of directly calling ben.getResponse("bye") keeps the logic encapsulated within MainWindow, maintaining separation of concerns.

SahanaAvantika committed:
The task update process was incorrectly handling task cloning, causing the past and present tasks to reference the same object.

This issue resulted in the "past" task being updated unexpectedly, leading to incorrect task display during updates.

Introduced the clone() method in concrete Task subclasses (e.g., Deadline, Event) to properly clone tasks before updates.

Without cloning, both the past and present tasks were referring to the same object, which caused incorrect task data to be shown in the UI.

SahanaAvantika committed:
update ReadMe

SahanaAvantika committed:
update Todo

SahanaAvantika committed:
update shadowJar

SahanaAvantika committed:
Update README.md

SahanaAvantika committed:
Update logo

SahanaAvantika committed:
Update scrollpanel and ReadMe

SahanaAvantika committed:
Merge tag 'A-JUnit'

# Conflicts:
#	data/ben.txt

SahanaAvantika committed:
Update test file

SahanaAvantika committed:
Update parser functions

SahanaAvantika committed:
Debug scrollpanel

SahanaAvantika committed:
Debug execute command

SahanaAvantika committed:
Add more Junit Testing

SahanaAvantika committed:
Edit JUnit test

SahanaAvantika committed:
Update Storage

SahanaAvantika committed:
Edit update command UI

SahanaAvantika committed:
Update ReadMe
